python/cugraph/traversal/bfs.py

```diff
@@ -191,34 +180,30 @@ def bfs(G,
     >>> df = cugraph.bfs(G, 0)
 
     """
-    (start, return_sp_counter, directed) = \
-        _ensure_args(G, start, return_sp_counter, i_start, directed)
+    (start, directed) = \
+        _ensure_args(G, start, i_start, directed)
 
     # FIXME: allow nx_weight_attr to be specified
     (G, input_type) = ensure_cugraph_obj(
         G, nx_weight_attr="weight",
         matrix_graph_type=DiGraph if directed else Graph)
 
-    if type(G) is Graph:
-        is_directed = False
-    else:
-        is_directed = True
-
     if G.renumbered is True:
-        start = G.lookup_internal_vertex_id(cudf.Series([start]))[0]
-
-    df = bfs_wrapper.bfs(G, start, is_directed, return_sp_counter)
+        if isinstance(start, cudf.DataFrame):
+            start = G.lookup_internal_vertex_id(start, start.columns).iloc[0]
+        else:
+            start = G.lookup_internal_vertex_id(cudf.Series([start]))[0]
 
+    df = bfs_wrapper.bfs(G, start, depth_limit, direction_optimizing)
     if G.renumbered:
         df = G.unrenumber(df, "vertex")
         df = G.unrenumber(df, "predecessor")
-        df["predecessor"].fillna(-1, inplace=True)
+        df.fillna(-1, inplace=True)
 
     return _convert_df_to_output_type(df, input_type)
 
 
-def bfs_edges(G, source, reverse=False, depth_limit=None, sort_neighbors=None,
-              return_sp_counter=False):
+def bfs_edges(G, source, reverse=False, depth_limit=None, sort_neighbors=None):
```

Comment:
Looks like `depth_limit` in the docstring still says it's not implemented.

Reply:
Its WIP, still need to clean up code

Reply:
updated doctstring